### ISSUE TITLE
[Bug]: --ignore-snapshots option doesn't work when combined with --ui mode

### ISSUE THREAD

**Issue by matewka** [microsoft/playwright#32868]
### Version

1.47.2 (tested to be present since 1.43.0)

### Steps to reproduce

1. Clone the repo: https://github.com/matewka/playwright-bug-test
2. Install dependencies with `yarn`
3. Run tests in CLI mode by executing: `yarn test`
4. Run the same tests in UI mode by executing: `yarn test:ui`

### Expected behavior

When Playwright is run with `--ignore-snapshots` flag, `expect(locator).toHaveScreenshot()` is always ignored, as described in the [docs](https://playwright.dev/docs/test-cli).

### Actual behavior

Test fails on snapshot comparison when `--ignore-snapshots` flag is used together with `--ui` flag.

### Environment

```shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 206.69 MB / 32.00 GB
  Binaries:
    Node: 18.19.1 - ~/.nvm/versions/node/v18.19.1/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 7.24.2 - /opt/homebrew/bin/npm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: 1.47.2 => 1.47.2
```

**Comment by Skn0tt** [comment]
Thanks for reporting, can reproduce. I'll write a fix.

**Comment by Skn0tt** [comment]
This will be fixed in the upcoming 1.48 release that's gonna be cut very soon.

**Comment by matewka** [comment]
Thank you very much for a blazing fast reaction 🤟
